torch/csrc/autograd/python_function.cpp

```diff
@@ -527,6 +527,42 @@ std::pair<UnpackedInput, InputFlags> unpack_input(PyObject *args) {
   return std::make_pair(std::move(unpacked), std::move(flags));
 }
 
+static void _append_subgraph(
```

Comment:
Consider adding comment explaining some context. Essentially answering
* Where is original subgraph?
* Why do you need `subgraph_trace_outputs` and `unpack_output` (consider renaming)?
* What happens to original subgraph?

Reply:
- unpack output is a variable to see whether if TupleUnpacking is needed